atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/AdjustableSweepBatchConfigSourceTest.java

```diff
@@ -0,0 +1,216 @@
+        //Given
+        configWithValues(1000, 1, 100);
+
+        whenDecreasingTheMultiplier_thenAdjustedConfigValuesDecrease();
+
+        whenIncreasingTheMultiplier_thenAdjustedConfigValuesIncrease();
+
+        assertThat(adjustableConfig.getBatchSizeMultiplier(), is(1.0));
+    }
+
+    private void whenIncreasingTheMultiplier_thenAdjustedConfigValuesIncrease() {
+        for (int i = 0; i < 1_000; i++) {
+            // When
+            adjustableConfig.increaseMultiplier();
+
+            // Then
+            batchSizeMultiplierIncreases();
+            batchSizeMultiplierDoesNotExceedOne();
+
+            maxCellTsPairsToExamineDoesNotDecrease();
+            candidateBatchSizeDoesNotDecrease();
+            deleteBatchSizeDoesNotDecrease();
+
+            maxCellTsPairsToExamineDoesNotExceedBaseConfig();
+            candidateBatchSizeDoesNotExceedBaseConfig();
+            deleteBatchSizeDoesNotExceedBaseConfig();
+
+            updatePreviousValues();
+        }
+    }
+
+    private void whenDecreasingTheMultiplier_thenAdjustedConfigValuesDecrease() {
+        for (int i = 0; i < 10_000; i++) {
+            // When
+            adjustableConfig.decreaseMultiplier();
+
+            // Then
+            batchSizeMultiplierDecreases();
+            maxCellTsPairsToExamineDecreasesToAMinimumOfOne();
+            candidateBatchSizeDecreasesToAMinimumOfOne();
+            deleteBatchSizeDecreasesToAMinimumOfOne();
+
+            updatePreviousValues();
+        }
+    }
+
+    private void configWithValues(int maxCellTsPairsToExamine, int candidateBatchSize, int deleteBatchSize) {
+        adjustableConfig = AdjustableSweepBatchConfigSource.create(() -> new SweepBatchConfig() {
+            @Override
+            public int maxCellTsPairsToExamine() {
+                return maxCellTsPairsToExamine;
+            }
+
+            @Override
+            public int candidateBatchSize() {
+                return candidateBatchSize;
+            }
+
+            @Override
+            public int deleteBatchSize() {
+                return deleteBatchSize;
+            }
+        });
+
+        updatePreviousValues();
+    }
+
+    private void batchSizeMultiplierDecreases() {
+        assertThat(adjustableConfig.getBatchSizeMultiplier(), is(lessThanOrEqualTo(previousMultiplier)));
+    }
+
+    private void maxCellTsPairsToExamineDecreasesToAMinimumOfOne() {
+        int newValue = adjustableConfig.getAdjustedSweepConfig().maxCellTsPairsToExamine();
+        if (newValue == 1) {
+            return;
+        }
+        assertThat(newValue, is(lessThan(previousMaxCellTsPairsToExamine)));
+    }
+
+    private void candidateBatchSizeDecreasesToAMinimumOfOne() {
+        int newValue = adjustableConfig.getAdjustedSweepConfig().candidateBatchSize();
+        if (newValue == 1) {
+            return;
+        }
+        assertThat(newValue, is(lessThan(previousCandidateBatchSize)));
```

Comment:
Nit: replace the three lines with:        `assertThat(newValue, is(anyOf(lessThan(previousMaxCellTsPairsToExamine), equalTo(1))));`

Same for the other two batch config value asserts.

Reply:
Done, had forgotten about `anyOf`